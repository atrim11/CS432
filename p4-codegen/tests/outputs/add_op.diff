--- outputs/add_op.txt	2024-11-13 13:31:23.953825918 -0500
+++ expected/add_op.txt	2024-11-01 14:38:22.832944811 -0400
@@ -1,15 +1,9 @@
 main:
-  push BP
-  i2i SP => BP
-  addI SP, 8 => SP
   loadI 2 => r0
   loadI 3 => r1
   add r0, r1 => r2
   i2i r2 => RET
-  jump l0
 l0:
-  i2i BP => SP
-  pop BP
   return
 
 ==========================
@@ -19,94 +13,49 @@
 other memory:
 ==========================
 
-Executing: push BP
-
-==========================
-sp=65528 bp=-9999999 ret=-9999999
-registers: 
-stack:  65528: -9999999
-other memory:
-==========================
-
-Executing: i2i SP => BP
-
-==========================
-sp=65528 bp=65528 ret=-9999999
-registers: 
-stack:  65528: -9999999
-other memory:
-==========================
-
-Executing: addI SP, 8 => SP
-
-==========================
-sp=65536 bp=65528 ret=-9999999
-registers: 
-stack:
-other memory:  65528: -9999999
-==========================
-
 Executing: loadI 2 => r0
 
 ==========================
-sp=65536 bp=65528 ret=-9999999
+sp=65536 bp=-9999999 ret=-9999999
 registers:  r0=2
 stack:
-other memory:  65528: -9999999
+other memory:
 ==========================
 
 Executing: loadI 3 => r1
 
 ==========================
-sp=65536 bp=65528 ret=-9999999
+sp=65536 bp=-9999999 ret=-9999999
 registers:  r0=2 r1=3
 stack:
-other memory:  65528: -9999999
+other memory:
 ==========================
 
 Executing: add r0, r1 => r2
 
 ==========================
-sp=65536 bp=65528 ret=-9999999
+sp=65536 bp=-9999999 ret=-9999999
 registers:  r0=2 r1=3 r2=5
 stack:
-other memory:  65528: -9999999
+other memory:
 ==========================
 
 Executing: i2i r2 => RET
 
 ==========================
-sp=65536 bp=65528 ret=5
-registers:  r0=2 r1=3 r2=5
-stack:
-other memory:  65528: -9999999
-==========================
-
-Executing: jump l0
-
-==========================
-sp=65536 bp=65528 ret=5
+sp=65536 bp=-9999999 ret=5
 registers:  r0=2 r1=3 r2=5
 stack:
-other memory:  65528: -9999999
-==========================
-
-Executing: i2i BP => SP
-
-==========================
-sp=65528 bp=65528 ret=5
-registers:  r0=2 r1=3 r2=5
-stack:  65528: -9999999
 other memory:
 ==========================
 
-Executing: pop BP
+Executing: l0:
 
 ==========================
 sp=65536 bp=-9999999 ret=5
 registers:  r0=2 r1=3 r2=5
 stack:
-other memory:  65528: -9999999
+other memory:
 ==========================
 
 Executing: return
