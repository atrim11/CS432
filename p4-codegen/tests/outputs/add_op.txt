main:
  push BP
  i2i SP => BP
  loadI 2 => r0
  loadI 3 => r1
  add r0, r1 => r2
  i2i r2 => RET
  jump l0
l0:
  i2i BP => SP
  pop BP
  return

==========================
sp=65536 bp=-9999999 ret=-9999999
registers: 
stack:
other memory:
==========================

Executing: push BP

==========================
sp=65528 bp=-9999999 ret=-9999999
registers: 
stack:  65528: -9999999
other memory:
==========================

Executing: i2i SP => BP

==========================
sp=65528 bp=65528 ret=-9999999
registers: 
stack:  65528: -9999999
other memory:
==========================

Executing: loadI 2 => r0

==========================
sp=65528 bp=65528 ret=-9999999
registers:  r0=2
stack:  65528: -9999999
other memory:
==========================

Executing: loadI 3 => r1

==========================
sp=65528 bp=65528 ret=-9999999
registers:  r0=2 r1=3
stack:  65528: -9999999
other memory:
==========================

Executing: add r0, r1 => r2

==========================
sp=65528 bp=65528 ret=-9999999
registers:  r0=2 r1=3 r2=5
stack:  65528: -9999999
other memory:
==========================

Executing: i2i r2 => RET

==========================
sp=65528 bp=65528 ret=5
registers:  r0=2 r1=3 r2=5
stack:  65528: -9999999
other memory:
==========================

Executing: jump l0

==========================
sp=65528 bp=65528 ret=5
registers:  r0=2 r1=3 r2=5
stack:  65528: -9999999
other memory:
==========================

Executing: i2i BP => SP

==========================
sp=65528 bp=65528 ret=5
registers:  r0=2 r1=3 r2=5
stack:  65528: -9999999
other memory:
==========================

Executing: pop BP

==========================
sp=65536 bp=-9999999 ret=5
registers:  r0=2 r1=3 r2=5
stack:
other memory:  65528: -9999999
==========================

Executing: return
RETURN VALUE = 5
