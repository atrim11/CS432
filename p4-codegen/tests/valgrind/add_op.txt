==344630== Memcheck, a memory error detector
==344630== Copyright (C) 2002-2022, and GNU GPL'd, by Julian Seward et al.
==344630== Using Valgrind-3.22.0 and LibVEX; rerun with -h for copyright info
==344630== Command: ../decaf inputs/add_op.decaf
==344630== 
main:
  loadI 2 => r0
  loadI 3 => r1
  add r0, r1 => r2
  i2i r2 => RET
l0:
  return

==========================
sp=65536 bp=-9999999 ret=-9999999
registers: 
stack:
other memory:
==========================

Executing: loadI 2 => r0

==========================
sp=65536 bp=-9999999 ret=-9999999
registers:  r0=2
stack:
other memory:
==========================

Executing: loadI 3 => r1

==========================
sp=65536 bp=-9999999 ret=-9999999
registers:  r0=2 r1=3
stack:
other memory:
==========================

Executing: add r0, r1 => r2

==========================
sp=65536 bp=-9999999 ret=-9999999
registers:  r0=2 r1=3 r2=5
stack:
other memory:
==========================

Executing: i2i r2 => RET

==========================
sp=65536 bp=-9999999 ret=5
registers:  r0=2 r1=3 r2=5
stack:
other memory:
==========================

Executing: l0:

==========================
sp=65536 bp=-9999999 ret=5
registers:  r0=2 r1=3 r2=5
stack:
other memory:
==========================

Executing: return
RETURN VALUE = 5
==344630== 
==344630== HEAP SUMMARY:
==344630==     in use at exit: 0 bytes in 0 blocks
==344630==   total heap usage: 1,814 allocs, 1,814 frees, 353,860 bytes allocated
==344630== 
==344630== All heap blocks were freed -- no leaks are possible
==344630== 
==344630== For lists of detected and suppressed errors, rerun with: -s
==344630== ERROR SUMMARY: 0 errors from 0 contexts (suppressed: 0 from 0)
