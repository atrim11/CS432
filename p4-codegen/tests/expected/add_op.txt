main:
  loadI 2 => r0
  loadI 3 => r1
  add r0, r1 => r2
  i2i r2 => RET
l0:
  return

==========================
sp=65536 bp=-9999999 ret=-9999999
registers: 
stack:
other memory:
==========================

Executing: loadI 2 => r0

==========================
sp=65536 bp=-9999999 ret=-9999999
registers:  r0=2
stack:
other memory:
==========================

Executing: loadI 3 => r1

==========================
sp=65536 bp=-9999999 ret=-9999999
registers:  r0=2 r1=3
stack:
other memory:
==========================

Executing: add r0, r1 => r2

==========================
sp=65536 bp=-9999999 ret=-9999999
registers:  r0=2 r1=3 r2=5
stack:
other memory:
==========================

Executing: i2i r2 => RET

==========================
sp=65536 bp=-9999999 ret=5
registers:  r0=2 r1=3 r2=5
stack:
other memory:
==========================

Executing: l0:

==========================
sp=65536 bp=-9999999 ret=5
registers:  r0=2 r1=3 r2=5
stack:
other memory:
==========================

Executing: return
RETURN VALUE = 5
